app/models/manageiq/providers/ansible_tower/shared/inventory/parser/automation_manager.rb

```diff
@@ -66,6 +68,14 @@ def configuration_script_sources
       inventory_object.status = project.status
       inventory_object.last_updated_on = project.last_updated
 
+      inventory_object.last_update_error = nil
+      unless project.status == 'successful'
+        last_update = project.last_update
+        if last_update
+          inventory_object.last_update_error = last_update.result_stdout[0..ERROR_MAX_SIZE]
```

Comment:
@Glutexo do you know if the stdout string comes as ASCII or can it be multibyte characters?

I'd prefer this to be changed to something like:
```ruby
inventory_object.last_update_error = last_update.result_stdout.mb_chars.limit(ERROR_MAX_SIZE)
```
(Maybe `to_s` would be needed at the end. I haven't tried to save it into DB yet.)

Reply:
Since it is a console output, it can definitely contain UTF-8 characters and I’d expect it to come as UTF-8 text. If it doesn’t, I’d consider it a bug. You can at least check the Ruby encoding of the string right before the save: `p(last_update.result_stdout.encoding)`. If it’s really UTF-8, then we can safely use `[0..n]`. Otherwise we’d have to investigate further what the real encoding is.

Reply:
I agree. In that case we shouldn't be using slices, since it counts letters, not bytes. 50kb != 50letters. The `limit` method on `mb_chars` is encoding safe, and limits the length to certain amount fitting the last character in (no matter how many bytes it is for single letter).

Reply:
You’re right. I was allured by the simplicity of `[]` so much, that I forgot that byte to char count does not map 1:1. (That me, who gives talks at conferences about character encodings and internationalization. What a shame! ☹️) Thanks for letting me know about [`limit`](http://api.rubyonrails.org/v5.0.6/)!